pom.xml

```diff
@@ -1622,5 +1623,62 @@
+      compile time. Auto-generated code is not checked.
+      See: https://errorprone.info/bugpatterns for list of available bug patterns.-->
+      <id>errorprone</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <configuration>
+              <showWarnings>true</showWarnings>
+              <source>${maven.compiler.source}</source>
+              <source>${maven.compiler.target}}</source>
+              <encoding>UTF-8</encoding>
+              <compilerArgs>
+                <arg>-XDcompilePolicy=simple</arg>
+                <arg>
+                  -Xplugin:ErrorProne \
+                  -XepExcludedPaths:.*/(proto|thrift|generated-sources)/.* \
+                  -XepDisableWarningsInGeneratedCode \
+                  -XepDisableAllWarnings \
+                  <!-- error/warning patterns to ignore -->
+                  -Xep:Incomparable:OFF \
+                  -Xep:CheckReturnValue:OFF \
+                  -Xep:MustBeClosedChecker:OFF \
+                  -Xep:ReturnValueIgnored:OFF \
+                  <!-- error/warning patterns to specifically check -->
+                  -Xep:ExpectedExceptionChecker \
+                  -Xep:MissingOverride \
+                  <!-- Items containing 'OFF' are currently flagged by errorprone. The 'OFF'
+                  can be removed and the project recompiled to discover lcation of errors for
+                  further analysis. @SuppressWarnings can be used to ignore errors if desired. -->
+                </arg>
+                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED</arg>
+                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED</arg>
+                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED</arg>
+                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED</arg>
+                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED</arg>
+                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED</arg>
+                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED</arg>
+                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED</arg>
+                <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED</arg>
+                <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED</arg>
+              </compilerArgs>
+              <annotationProcessorPaths>
+                <path>
+                  <groupId>com.google.errorprone</groupId>
+                  <artifactId>error_prone_core</artifactId>
+                  <version>${errorprone.version}</version>
+                </path>
+              </annotationProcessorPaths>
```

Comment:
When you set the annotation processor paths explicitly, does it prevent the automatic annotation processors from being picked up for the auto-service?

Reply:
After reading the docs a little more, it looks like this would cause issues if other annotations processor are used. It would no longer scan the compile classpath for annotation processors. They would have to be added using the  annotationProcessorPaths argument. 

Given that could be an issue, I'm ok closing this ticket without merging as it would be easy enough for me to run the processor every month or so on my own to see if any uncaught issues have been introduced into the codebase.

Reply:
Couldn't you just add this artifact as an `<optional>true</optional>` dependency in this profile, so it gets picked up alongside the other annotation processors, rather than being explicitly set and overriding the others?

Reply:
Or, you could modify our other annotation processors to be explicitly run like you did with this one, rather than used as a dependency. Then these configs could be merged when the profile is activated.

Reply:
Does it look like this is feasible to include? I'm finding this tool helpful.

Reply:
@DomGarguilo I've upgraded the version from 1.11 to 1.12. I added an additional exception so that non-ASCII Unicode in method names is not treated as an error. I also added the spotbugs annotation processor into the profile. If @ctubbsii can take a look and see if the POM is viable I think it can be included. 

For the moment, I would prefer to keep it as a profile rather than directly adding into the default build until I can better understand if it is using a separate java compiler. The documentation does make that distinction clear - at least as far as I can find so far. Also, adding to the default build will take some work to address various warnings so that the profile will not become overly large due to its aggresive nature. As it is set up now, errors are found unless specifically turned off. But warnings are off by default unless specifically activated. We can add specific warnings of interest as we go forward.